flink-clients/src/test/java/org/apache/flink/client/deployment/application/ApplicationDispatcherBootstrapITCase.java

```diff
@@ -106,8 +112,16 @@ public void testDispatcherRecoversAfterLosingAndRegainingLeadership() throws Exc
                     JobStatus.RUNNING,
                     deadline);
 
+            // make sure the operator is actually running
+            BlockingJob.awaitRunning(blockId);
+
+            final CompletableFuture<JobResult> firstJobResult =
+                    cluster.requestJobResult(ApplicationDispatcherBootstrap.ZERO_JOB_ID);
             // revoke & re-grant dispatcher leadership
             haServices.revokeDispatcherLeadership();
+            // make sure the leadership is revoked to avoid race conditions
+            Assertions.assertEquals(
```

Comment:
How does adding an assert resolve a race condition?

Reply:
The important part here is calling the `get()` method on the job result future, which blocks until the application "completes" / leadership is revoked.

The assert itself just checks that the returned status is as we expect it to be.

Reply:
Sounds good!